### ISSUE TITLE
Truffle fails to compile with jdk7

### ISSUE THREAD

**Issue by dougxc** [graalvm/truffle#128]
On a clean Truffle repo:

```
dsimon@kurz ~/g/truffle> mx --strict build -n --no-javac-crosscompile --dependencies com.oracle.truffle.api.interop.java
Compiling com.oracle.truffle.api with javac... [/Users/dsimon/graal/truffle/mxbuild/truffle/com.oracle.truffle.api/bin/com/oracle/truffle/api/Assumption.class does not exist]
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/instrument/impl/DefaultVisualizer.java:91: warning: [deprecation] displayIdentifier(FrameSlot) in Visualizer has been deprecated
    public String displayIdentifier(FrameSlot slot) {
                  ^
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/instrument/impl/DefaultVisualizer.java:83: warning: [deprecation] displayValue(Object,int) in Visualizer has been deprecated
    public String displayValue(Object value, int trim) {
                  ^
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/instrument/impl/DefaultVisualizer.java:78: warning: [deprecation] displayCallTargetName(CallTarget) in Visualizer has been deprecated
    public String displayCallTargetName(CallTarget callTarget) {
                  ^
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/instrument/impl/DefaultVisualizer.java:66: warning: [deprecation] displayMethodName(Node) in Visualizer has been deprecated
    public String displayMethodName(Node node) {
                  ^
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/instrument/impl/DefaultVisualizer.java:49: warning: [deprecation] displaySourceLocation(Node) in Visualizer has been deprecated
    public String displaySourceLocation(Node node) {
                  ^
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/instrument/impl/DefaultVisualizer.java:44: warning: [deprecation] getASTPrinter() in Visualizer has been deprecated
    public ASTPrinter getASTPrinter() {
                      ^
6 warnings
Compiling com.oracle.truffle.api.dsl with javac... [dependency com.oracle.truffle.api updated]
Compiling com.oracle.truffle.api.interop with javac... [dependency com.oracle.truffle.api updated]
Compiling com.oracle.truffle.api.instrumentation with javac... [dependency com.oracle.truffle.api updated]
Compiling com.oracle.truffle.dsl.processor with javac... [dependency com.oracle.truffle.api.dsl updated]
Compiling com.oracle.truffle.api.object with javac... [dependency com.oracle.truffle.api.interop updated]
Compiling com.oracle.truffle.api.object.dsl with javac... [dependency com.oracle.truffle.api.object updated]
Compiling com.oracle.truffle.object.dsl.processor with javac... [dependency com.oracle.truffle.api.object.dsl updated]
Archiving TRUFFLE_DSL_PROCESSOR_INTERNAL... [dependency com.oracle.truffle.dsl.processor updated]
Compiling com.oracle.truffle.api.interop.java with javac... [dependency TRUFFLE_DSL_PROCESSOR_INTERNAL updated]
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ArrayGetSizeNode.java:33: error: cannot find symbol
final class ArrayGetSizeNode extends ArrayGetSizeBaseNode {
                                     ^
  symbol: class ArrayGetSizeBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ArrayHasSizeNode.java:33: error: cannot find symbol
final class ArrayHasSizeNode extends ArrayHasSizeBaseNode {
                                     ^
  symbol: class ArrayHasSizeBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ExecuteNode.java:39: error: cannot find symbol
final class ExecuteNode extends JavaFunctionBaseNode {
                                ^
  symbol: class JavaFunctionBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/InvokeNode.java:36: error: cannot find symbol
final class InvokeNode extends InvokeBaseNode {
                               ^
  symbol: class InvokeBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/IsExecutableNode.java:31: error: cannot find symbol
final class IsExecutableNode extends IsExecutableBaseNode {
                                     ^
  symbol: class IsExecutableBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/NewNode.java:34: error: cannot find symbol
final class NewNode extends JavaNewBaseNode {
                            ^
  symbol: class JavaNewBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/NullCheckNode.java:31: error: cannot find symbol
final class NullCheckNode extends NullCheckBaseNode {
                                  ^
  symbol: class NullCheckBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ReadFieldNode.java:40: error: cannot find symbol
final class ReadFieldNode extends ReadFieldBaseNode {
                                  ^
  symbol: class ReadFieldBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/WriteFieldNode.java:33: error: cannot find symbol
final class WriteFieldNode extends WriteFieldBaseNode {
                                   ^
  symbol: class WriteFieldBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ExecuteNode.java:41: error: cannot find symbol
    @Child private DoExecuteNode doExecute;
     ^
  symbol:   class Child
  location: class ExecuteNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/InvokeNode.java:38: error: cannot find symbol
    @Child private DoExecuteNode doExecute;
     ^
  symbol:   class Child
  location: class InvokeNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/WriteFieldNode.java:35: error: cannot find symbol
    @Child private ToJavaNode toJava = new ToJavaNode();
     ^
  symbol:   class Child
  location: class WriteFieldNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ExecuteNode.java:41: error: Exception thrown during processing: java.lang.ClassCastException: com.sun.tools.javac.code.Symbol$VarSymbol cannot be cast to javax.lang.model.element.ExecutableElement
    @Child private DoExecuteNode doExecute;
                                 ^
  	at com.oracle.truffle.dsl.processor.verify.VerifyTruffleProcessor.process(VerifyTruffleProcessor.java:109)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:793)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:722)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1700(JavacProcessingEnvironment.java:97)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1029)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1163)
  	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1108)
  	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:824)
  	at com.sun.tools.javac.main.Main.compile(Main.java:439)
  	at com.sun.tools.javac.main.Main.compile(Main.java:353)
  	at com.sun.tools.javac.main.Main.compile(Main.java:342)
  	at com.sun.tools.javac.main.Main.compile(Main.java:333)
  	at com.sun.tools.javac.Main.compile(Main.java:76)
  	at com.sun.tools.javac.Main.main(Main.java:61)
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/InvokeNode.java:38: error: Exception thrown during processing: java.lang.ClassCastException: com.sun.tools.javac.code.Symbol$VarSymbol cannot be cast to javax.lang.model.element.ExecutableElement
    @Child private DoExecuteNode doExecute;
                                 ^
  	at com.oracle.truffle.dsl.processor.verify.VerifyTruffleProcessor.process(VerifyTruffleProcessor.java:109)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:793)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:722)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1700(JavacProcessingEnvironment.java:97)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1029)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1163)
  	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1108)
  	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:824)
  	at com.sun.tools.javac.main.Main.compile(Main.java:439)
  	at com.sun.tools.javac.main.Main.compile(Main.java:353)
  	at com.sun.tools.javac.main.Main.compile(Main.java:342)
  	at com.sun.tools.javac.main.Main.compile(Main.java:333)
  	at com.sun.tools.javac.Main.compile(Main.java:76)
  	at com.sun.tools.javac.Main.main(Main.java:61)
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/WriteFieldNode.java:35: error: Exception thrown during processing: java.lang.ClassCastException: com.sun.tools.javac.code.Symbol$VarSymbol cannot be cast to javax.lang.model.element.ExecutableElement
    @Child private ToJavaNode toJava = new ToJavaNode();
                              ^
  	at com.oracle.truffle.dsl.processor.verify.VerifyTruffleProcessor.process(VerifyTruffleProcessor.java:109)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:793)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:722)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1700(JavacProcessingEnvironment.java:97)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1029)
  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1163)
  	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1108)
  	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:824)
  	at com.sun.tools.javac.main.Main.compile(Main.java:439)
  	at com.sun.tools.javac.main.Main.compile(Main.java:353)
  	at com.sun.tools.javac.main.Main.compile(Main.java:342)
  	at com.sun.tools.javac.main.Main.compile(Main.java:333)
  	at com.sun.tools.javac.Main.compile(Main.java:76)
  	at com.sun.tools.javac.Main.main(Main.java:61)
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ArrayGetSizeNode.java:33: error: cannot find symbol
final class ArrayGetSizeNode extends ArrayGetSizeBaseNode {
                                     ^
  symbol: class ArrayGetSizeBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ArrayHasSizeNode.java:33: error: cannot find symbol
final class ArrayHasSizeNode extends ArrayHasSizeBaseNode {
                                     ^
  symbol: class ArrayHasSizeBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ExecuteNode.java:39: error: cannot find symbol
final class ExecuteNode extends JavaFunctionBaseNode {
                                ^
  symbol: class JavaFunctionBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/InvokeNode.java:36: error: cannot find symbol
final class InvokeNode extends InvokeBaseNode {
                               ^
  symbol: class InvokeBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/IsExecutableNode.java:31: error: cannot find symbol
final class IsExecutableNode extends IsExecutableBaseNode {
                                     ^
  symbol: class IsExecutableBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/NewNode.java:34: error: cannot find symbol
final class NewNode extends JavaNewBaseNode {
                            ^
  symbol: class JavaNewBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/NullCheckNode.java:31: error: cannot find symbol
final class NullCheckNode extends NullCheckBaseNode {
                                  ^
  symbol: class NullCheckBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ReadFieldNode.java:40: error: cannot find symbol
final class ReadFieldNode extends ReadFieldBaseNode {
                                  ^
  symbol: class ReadFieldBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/WriteFieldNode.java:33: error: cannot find symbol
final class WriteFieldNode extends WriteFieldBaseNode {
                                   ^
  symbol: class WriteFieldBaseNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/ExecuteNode.java:41: error: cannot find symbol
    @Child private DoExecuteNode doExecute;
     ^
  symbol:   class Child
  location: class ExecuteNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/InvokeNode.java:38: error: cannot find symbol
    @Child private DoExecuteNode doExecute;
     ^
  symbol:   class Child
  location: class InvokeNode
/Users/dsimon/graal/truffle/truffle/com.oracle.truffle.api.interop.java/src/com/oracle/truffle/api/interop/java/WriteFieldNode.java:35: error: cannot find symbol
    @Child private ToJavaNode toJava = new ToJavaNode();
     ^
  symbol:   class Child
  location: class WriteFieldNode
15 errors
```

**Comment by cosminbasca** [comment]
I can confirm this, works on jdk8.

**Comment by dougxc** [comment]
This is fixed by #136.
